tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1363,16 +1364,18 @@ protected:
       auto newItemField = fSubFields[0]->Clone(fSubFields[0]->GetName());
       return std::make_unique<RField<ROOT::VecOps::RVec<ItemT>>>(newName, std::move(newItemField));
    }
-   void AppendImpl(const Detail::RFieldValue& value) final {
+   std::size_t AppendImpl(const Detail::RFieldValue& value) final {
       auto typedValue = value.Get<ContainerT>();
+      auto nbytes = 0;
       auto count = typedValue->size();
       for (unsigned i = 0; i < count; ++i) {
          auto itemValue = fSubFields[0]->CaptureValue(&typedValue->data()[i]);
-         fSubFields[0]->Append(itemValue);
+         nbytes += fSubFields[0]->Append(itemValue);
       }
       Detail::RColumnElement<ClusterSize_t, EColumnType::kIndex> elemIndex(&fNWritten);
       fNWritten += count;
       fColumns[0]->Append(elemIndex);
+      return nbytes + sizeof(ClusterSize_t);
```

Comment:
The `nbytes` of data and the count in `elemIndex` are written to different columns. The return value indicates the total amount of bytes written so that we can keep track of the uncompressed size of the current cluster.